src/Network/Network/AzureFirewallPolicy/RuleGroup/NewAzureFirewallPolicyRuleCollectionGroupCommand.cs

```diff
@@ -0,0 +1,113 @@
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Network.Models;
+using MNM = Microsoft.Azure.Management.Network.Models;
+using Microsoft.Azure.Management.Network;
+using Newtonsoft.Json;
+using Microsoft.Rest.Azure;
+using Microsoft.Rest.Serialization;
+using System.IO;
+using Newtonsoft.Json.Linq;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "FirewallPolicyRuleCollectionGroup", SupportsShouldProcess = true, DefaultParameterSetName = SetByNameParameterSet), OutputType(typeof(PSAzureFirewallPolicyRuleCollectionGroup))]
+    public class NewAzureFirewallPolicyRuleCollectionGroupCommand : AzureFirewallPolicyRuleCollectionGroupBaseCmdlet
+    {
+
+        private const string SetByNameParameterSet = "SetByNameParameterSet";
+        private const string SetByInputObjectParameterSet = "SetByInputObjectParameterSet";
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The name of the Rule Group")]
+        [ValidateNotNullOrEmpty]
+        public virtual string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The priority of the rule group")]
+        [ValidateRange(100, 65000)]
+        public uint Priority { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The list of rule collections")]
+        [ValidateNotNullOrEmpty]
+        public PSAzureFirewallPolicyBaseRuleCollection[] RuleCollection { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name.")]
+        [ValidateNotNullOrEmpty]
+        public virtual string ResourceGroupName { get; set; }
```

Comment:
I have a question here @isra-fel . As this is the resourcegroupname of the parent, should I modify this to FirewallPolicyResourceGroupName? Wouldn't the user be confused and enter some other resourcegroup name with the current naming and this call fails?

Reply:
You don't need to. It's a common pattern in every azure powershell cmdelt. Don't worry :)